lang/csharp/src/apache/main/AvroDecimal.cs

```diff
@@ -0,0 +1,983 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Globalization;
+using System.Numerics;
+
+namespace Avro
+{
+    /// <summary>
+    /// Represents a big decimal.
+    /// </summary>
+    #pragma warning disable CS1591 // Missing XML comment for publicly visible type or member
+    public struct AvroDecimal : IConvertible, IFormattable, IComparable, IComparable<AvroDecimal>, IEquatable<AvroDecimal>
+    {
+        /// <summary>
+        /// The number '-1'.
+        /// </summary>
+        public static readonly AvroDecimal MinusOne = new AvroDecimal(BigInteger.MinusOne, 0);
```

Comment:
Do you think the `MinusOne`, `Zero`, `One`, `IsEven`, `IsOne`, `IsPowerOfTwo`, `IsZero` and `Sign` properties are useful? (I can't see the value in mimicking the BigInteger class here)

Reply:
`ONE`, `ZERO`, and `TEN` are available on Java's `BigDecimal` class. You could lose the `IsZero` and `IsOne` properties because you could easily compare against `Zero` or `One` (i.e., `myDec == AvroDecimal.Zero`), or even `myDec == 0.0m` which will select the `==` operator.

I'm not precious about any of these properties.